src/org/omegat/util/StaticUtils.java

```diff
@@ -579,4 +580,19 @@ public static boolean isProjectDir(File f) {
         return true;
     }
 
+    public static String getFirstLine(InputStream stream, int maxLength) throws IOException {
+        byte[] data = new byte[maxLength];
+        int read = stream.read(data);
+        if (isAsciiBytes(data, read)) {
+            return new String(data, StandardCharsets.US_ASCII);
+        }
+        return null;
+    }
+
+    private static boolean isAsciiBytes(byte[] data, int size) {
+        for (int i = 0; i < size; i++) {
+            if (data[i] < 0) return false;
+        }
+        return true;
+    }
```

Comment:
I think it's a lot easier to use a `Reader` to read in the first line. Is there a reason you went with this implementation instead?

Also I think you're not buffering the reading, so this might be hitting the filesystem up to `maxLength` times(?).

Reply:
When using Reader, spotBugs warns it is bad when calling Reader without Encoding.
Here is a place to detect encoding, so I read it as bytes.

Reply:
> Here is a place to detect encoding, so I read it as bytes.

But ultimately you didn't read it as bytes: you read it as an ASCII string. That's the same as using a `Reader` with ASCII encoding.